File: src/entities/author.entity.ts
import { Field, ObjectType } from "type-graphql";
import {
  BaseEntity,
  Column,
  CreateDateColumn,
  DeleteDateColumn,
  Entity,
  OneToMany,
  PrimaryColumn,
  UpdateDateColumn,
} from "typeorm";
import { Blog } from "./blog.entity";

@ObjectType()
@Entity()
export class Author extends BaseEntity {
  @Field()
  @PrimaryColumn("varchar", { length: 255 })
  email: string;

  @Field()
  @Column("varchar", { length: 255 })
  username: string;

  @Field()
  @Column("varchar", { length: 255, nullable: true })
  image: string;

  @Field()
  @Column("text", { nullable: true })
  description: string;

  @Field()
  @CreateDateColumn()
  createdAt: Date;

  @Field()
  @UpdateDateColumn()
  updatedAt: Date;

  @Field()
  @DeleteDateColumn()
  deletedAt: Date;

  @Field(() => [Blog])
  @OneToMany(() => Blog, (blog) => blog.author)
  blogs: Blog[];
}

File: internal/entities/author_entity.go
package entities

type Author struct {
	Id          string `json:"id"`
	Email       string `json:"email"`
	Username    string `json:"username"`
	Image       string `json:"image,omitempty"`
	Description string `json:"description,omitempty"`
	CreatedAt   string `json:"createdAt,omitempty"`
	UpdatedAt   string `json:"updatedAt,omitempty"`
	DeletedAt   string `json:"deletedAt,omitempty"`
	Blogs       []Blog `json:"blogs"`
}
